codec-sockjs/src/main/java/io/netty/handler/codec/sockjs/util/JsonConverter.java

```diff
@@ -0,0 +1,269 @@
+     */
+    public static ByteBuf escapeJson(final ByteBuf input, final ByteBuf buffer) {
+        final int count = input.readableBytes();
+        for (int i = 0; i < count; i++) {
+            final byte ch = input.getByte(i);
+            switch(ch) {
+                case '"':
+                    buffer.writeByte('\\').writeByte('\"');
+                    break;
+                case '/':
+                    buffer.writeByte('\\').writeByte('/');
+                    break;
+                case '\\':
+                    buffer.writeByte('\\').writeByte('\\');
+                    break;
+                case '\b':
+                    buffer.writeByte('\\').writeByte('b');
+                    break;
+                case '\f':
+                    buffer.writeByte('\\').writeByte('f');
+                    break;
+                case '\n':
+                    buffer.writeByte('\\').writeByte('n');
+                    break;
+                case '\r':
+                    buffer.writeByte('\\').writeByte('r');
+                    break;
+                case '\t':
+                    buffer.writeByte('\\').writeByte('t');
+                    break;
+                default:
+                    if (shouldEscape((char) ch)) {
+                        final String hex = Integer.toHexString(ch);
+                        buffer.writeByte('\\').writeByte('u');
+                        for (int k = 0; k < 4 - hex.length(); k++) {
+                            buffer.writeByte('0');
+                        }
+                        buffer.writeBytes(hex.toLowerCase().getBytes());
+                    } else {
+                        buffer.writeByte(ch);
+                    }
+            }
+        }
+        return buffer;
+    }
+
+    public static boolean isControlCharacter(final char c) {
+        return c >= '\u0000' && c <= '\u001F';
+    }
+
+    public static boolean isSurrogateCharacter(final char c) {
+        return c >= Character.MIN_HIGH_SURROGATE && c <= Character.MAX_LOW_SURROGATE;
+    }
+
+    public static boolean isFormatControlCharacter(final char c) {
+        return c == '\u200C' || c == '\u200D';
+    }
+
+    public static boolean isSeparatorCharacter(final char c) {
+        return c >= '\u2028' && c <= '\u202F';
+    }
+
+    public static boolean isFormatOtherCharacter(final char c) {
+        return c >= '\u2060' && c <= '\u206F';
+    }
+
+    public static boolean isSpecials(final char c) {
+        return c >= '\uFFF0' && c <= '\uFFFF';
+    }
+
+    /**
+     * Processes the input char[] array and escapes the any control characters, quotes, slashes,
+     * and unicode characters.
+     *
+     * @param value the char[] for which unicode characters should be escaped
+     * @return {@code String} Java style escaped unicode characters.
+     */
+    public static String escapeCharacters(final char[] value) {
```

Comment:
![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png) Consider using varargs for methods or constructors which take an array the last parameter. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://garage.netty.io/sonarqube/coding_rules#rule_key=pmd%3AUseVarargs)